libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -823,6 +840,34 @@ void CompilerUtils::cleanHigherOrderBits(IntegerType const& _typeOnStack)
 		m_context << ((u256(1) << _typeOnStack.numBits()) - 1) << Instruction::AND;
 }
 
+void CompilerUtils::cleanHigherOrderBits(FixedPointType const& _typeOnStack)
+{
+	if (_typeOnStack.numBits() == 256)
+		return;
+	else if (_typeOnStack.isSigned())
+		m_context << u256(_typeOnStack.numBits() / 8 - 1) << Instruction::SIGNEXTEND;
+	else
+		m_context << ((u256(1) << _typeOnStack.numBits()) - 1) << Instruction::AND;
+}
+
+void CompilerUtils::fixedPointShifting(FixedPointType const& _stackType, FixedPointType const& _targetType)
```

Comment:
it's a helper function. It works for two functions. I felt the duplication was unnecessary.